extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -48,73 +49,96 @@
+      logger.info( Messages.getInstance().getString( "ActionInvokerSystemListener.INFO_0001_READ_FILES_FROM_FOLDER", environmentVariablesFolder, WORK_ITEM_FILE_EXTENSION ) );
     } else if ( !StringUtils.isEmpty( solutionPath ) ) {
       files = listFiles( new File( solutionPath ), WORK_ITEM_FILE_EXTENSION );
-      logger.info( "Reading " + WORK_ITEM_FILE_EXTENSION + " files from " + solutionPath );
+      logger.info( Messages.getInstance().getString( "ActionInvokerSystemListener.INFO_0003_READ_DATA_FROM_FILE", WORK_ITEM_FILE_EXTENSION, solutionPath ) );
     }
 
     if ( files == null || files.length == 0 ) {
-      logger.info( "No " + WORK_ITEM_FILE_EXTENSION + " files found. Exiting." );
+      logger.info( Messages.getInstance().getString( "ActionInvokerSystemListener.INFO_0002_NO_FILES_FOUND", WORK_ITEM_FILE_EXTENSION ) );
+      placeBreadcrumbFile( getSolutionPath(), false );
       return true;
     }
 
     for ( File file : files ) {
-      logger.info( "Attempting to read data from " + file.getAbsolutePath() );
+      logger.info( Messages.getInstance().getString( "ActionInvokerSystemListener.INFO_0003_READ_DATA_FROM_FILE", file.getAbsolutePath() ) );
+      FileInputStream fileInputStream = null;
       try {
-        FileInputStream fileInputStream = new FileInputStream( file );
+        fileInputStream = new FileInputStream( file );
         String encoded = IOUtils.toString( fileInputStream );
-        fileInputStream.close();
         Payload payload = new Payload( encoded, LocaleHelper.UTF_8 );
-        logger.info( "Issuing Work Item request" );
+        logger.info( Messages.getInstance().getString( "ActionInvokerSystemListener.INFO_0004_ISSUE_REQUEST" ) );
         try {
           IActionInvokeStatus status = issueRequest( payload );
+          if ( status != null && status.getThrowable() == null ) {
+            logger.info( Messages.getInstance().getString( "ActionInvokerSystemListener.INFO_0005_REQUEST_SUCCEEDED"  ) );
+            workItemSuccess = true;
+          } else {
+            logger.error( Messages.getInstance().getErrorString( "ActionInvokerSystemListener.ERROR_0001_REQUEST_FAILED" ) );
+          }
         } catch ( Exception e ) {
-          //send status to chronos
+          logger.error( e.getMessage() );
         }
       } catch ( IOException e ) {
-        logger.error( "Error reading files. " );
+        logger.error( Messages.getInstance().getErrorString( "ActionInvokerSystemListener_ERROR_0002_ERROR_READING_FILES" ) );
       } catch ( IllegalArgumentException e ) {
-        logger.error( file.getName() + " Payload could not be recreated. Will not process." );
+        logger.error( Messages.getInstance().getErrorString( "ActionInvokerSystemListener_ERROR_0003_COULD_NOT_PROCESS", file.getName() ) );
       } finally {
-        renameFile( file.getAbsolutePath() );
+        if ( fileInputStream != null ) {
```

Comment:
extensions has a FileUtils class - maybe this could be added as a method called closeQuietly( FileInputStream fs) or something